### PR TITLE
extend checkpoint structure to include lsig address.

### PR DESCRIPTION
**Pull request by amityadav0**
## Proposed Changes

+ extend checkpoint structure to include lsig address.
+ update the checkpoint guide to explain lsig checkpoints
+ Fix linter errors

### CODE REVIEW

File: docs/guide/execution-checkpoints.md [hunk 1]
```diff
@@ -74,6 +74,27 @@ Consider a case when App is updated with same programs, this is the case where w
...
 While retreiving AppInfo from checkpoints only the lastest timestamp AppInfo is used.
 
+## Checkpoint storage for logic signature(Lsig)
+
+- For logic signature in contract mode: You may use `addLsigCheckpoint` function to store lsigInfo in checkpoint.
+- To load logic signature in contract mode: You may use `getContractLsig` function.
+- Ex:
+```js
+// store checkpoint
+deployer.addLsigCheckpoint(contractName);
```
Comment (ratik21) [inline docs/guide/execution-checkpoints.md:84]: Why not use/expose `mkContractLsig`?

File: examples/permissioned-voting/scripts/vote/result.js [hunk 2]
```diff
@@ -26,24 +26,24 @@ async function run (runtimeEnv, deployer) {
...
-  // // Delete Application
-  // console.log('Deleting Application');
-  // await executeTransaction(deployer, txnParam);
-
-  // txnParam.fromAccount = alice;
-  // txnParam.type = types.TransactionType.ClearApp;
-
-  // // Clear voter's account
-  // console.log("Clearing Alice's Account");
-  // await executeTransaction(deployer, txnParam);
```
Comment (ratik21) [inline examples/permissioned-voting/scripts/vote/result.js:46]: Why was code commented before?

File: packages/algob/src/internal/deployer.ts [hunk 3]
```diff
@@ -153,6 +153,13 @@ class DeployerBasicMode {
     return lsigAccount;
   }
 
+  /**
+   * Loads a logic signature account from checkpoint
+  */
```
Comment (ratik21) [inline packages/algob/src/internal/deployer.ts:158]: please add @param in description as well

File: packages/algob/src/internal/deployer.ts [hunk 4]
```diff
@@ -586,6 +593,31 @@ export class DeployerDeployMode extends DeployerBasicMode implements Deployer {
...
+      };
+    } catch (error) {
+      this.persistCP();
+
+      console.log(error);
+      throw error;
+    }
+    this.cpData.registerLsig(this.networkName, name, lsigInfo);
+    return lsigInfo;
+  }
```
Comment (ratik21) [inline packages/algob/src/internal/deployer.ts:619]: I think most of  the code here is same with `mkDelegatedLsig` (except `lsig.sign`). Can we move code to a common function (maybe `_mkLsig`) & reuse in both?

Comment (amityadav0) [reply packages/algob/src/internal/deployer.ts:619]: how do we remove `lsig.sign` in a common function ?

File: packages/algob/src/internal/deployer.ts (hunk 3 above)
Comment (amityadav0) [reply packages/algob/src/internal/deployer.ts:158]: ok

File: examples/permissioned-voting/scripts/vote/result.js (hunk 2 above)
Comment (amityadav0) [reply examples/permissioned-voting/scripts/vote/result.js:46]: i don't know

File: docs/guide/execution-checkpoints.md (hunk 1 above)
Comment (amityadav0) [reply docs/guide/execution-checkpoints.md:84]: ok let's use this

File: packages/algob/src/internal/deployer.ts (hunk 4 above)
Comment (ratik21) [reply packages/algob/src/internal/deployer.ts:619]: I was thinking `_mkLsig(lsig, signer?)`. If signer is passed we store delegated lsig. Could something like this work? Could you please check?

File: examples/permissioned-voting/scripts/vote/result.js (hunk 2 above)
Comment (robert-zaremba) [reply examples/permissioned-voting/scripts/vote/result.js:46]: It seams that @amityadav0 commented it out last week. Maybe some mistake?

Comment (amityadav0) [reply examples/permissioned-voting/scripts/vote/result.js:46]: yes

File: packages/algob/src/internal/deployer.ts (hunk 4 above)
Comment (amityadav0) [reply packages/algob/src/internal/deployer.ts:619]: ok

**Review by ratik21 — APPROVED** [review]

**Pull request merged**